### DISCUSSION TITLE
Overlay a running indicator for frozen video detection

### DISCUSSION THREAD

**Discussion by toliuweijing** [izivkov/RemoteVideoCam#24] [Ideas]
**Background**

<img width="670" alt="image" src="https://github.com/user-attachments/assets/862f125f-f111-44e4-af94-e89f2335fa84">

As shown above, it is suggested to place a constant moving object in the scene to detect the frozen video.

In v3.3, motion detection is added to detect movements in the video frame. This feature doesn't work well if the scene has a moving object. 

**Proposal**
One idea is to overlay a running indicator in the video stream, e.g. a blinking dot, timestamp watermark. This approach removes the needs of placing a moving object. User can easily check the running indicator for frozen video, while the motion detection can function as expected.

**Comment by izivkov** [comment]
Yes, I think this is a good idea. Feel free to add it the existing PR.
Also, feel free to update the README file.
Regards

**Comment by toliuweijing** [comment]
Ran into frozen video last night. Will try to implement a prototype today.

**Comment by izivkov** [comment]
Awesome, thanks for the update!

**Comment by toliuweijing** [comment]
I have a demo. It provides two functionalities related to this and [another one](https://github.com/izivkov/RemoteVideoCam/discussions/26).

1. provides a button to turn off live camera stream for pure security camera usage, saving heat and battery.
2. provides an view-base indicator to detect if the video stream is stuck for more than 1 second. This is more performance than overlaying an indicator into the camera stream via OpenGL, which introduces more GPU work and slowing frame rate.

|camera toggle| froze video indicator
|-|-
|![output](https://github.com/user-attachments/assets/796aec11-473a-48b1-9066-e9fcdc947ff9)|![output](https://github.com/user-attachments/assets/b9e6d480-2fad-4272-9704-c9fbb34fb469)
|User can turn off the video stream by toggling the videocam button. It is useful for pure security cam use case where turning off camera avoids battery consumption and heat |A indicator detects video stream status and turns red after being stuck for more than 1 second. It turns green when videos stream recovers.


@izivkov WDYT

**Comment by izivkov** [comment]
I think the red dot for video freeze is great. I think I would make the dot
a bit bigger, it is hard to notice.

I think for stopping the stream manually, people would often forget to do
it. For myself, I would prefer to turn the camera on after a motion
detection. Turn it off otherwise. I realize this may not work if the camera
is pointing to a busy street, it will always be on. Your method will work
as well, so let us think about this.

I'm also wondering if the overheating issues is not particular to the Pixel
phone. They have tensor chips to process video, so this might be causing
it. I general don't have the issue with my cheap Motorola phone.

**Reply by izivkov** [reply]
@toliuweijing 
Jim, I got this message from a user:

--------------------------------------------------------
https://github.com/user-attachments/assets/3b89f32b-69a9-4b5b-9151-a3fb2e9931e5

How do you explain 5 TIMES bigger size of this app on new update???
----------------------------------------------------------------------------

Looks like a lot of exctra size just for detecting motion. Did we add something we do not need? Can you take a look?

Thanks

**Reply by toliuweijing** [reply]
Sure. Let me investigate. Maybe some proguard configuration issues causes OpenCV not being stripped in release build.
